core/src/test/resources/org/apache/calcite/test/RelOptRulesTest.xml

```diff
@@ -5397,22 +5397,25 @@ where d.deptno > 7 and e.deptno > 9]]>
         <Resource name="planBefore">
             <![CDATA[
 LogicalProject(EXPR$0=[1])
-  LogicalFilter(condition=[>($16, 9)])
```

Comment:
This changes
```sql
select 1 from sales.emp d
left outer join sales.emp e on d.deptno = e.deptno
where d.deptno > 7 and e.deptno > 9
```
to 
```sql
select 1 from sales.emp d
inner join sales.emp e on d.deptno = e.deptno
where d.deptno > 7 and e.deptno > 9
```

Reply:
The change is valid.

Reply:
Out of curiosity: does this change imply that inner join is always better than outer join when they are equivalent (e.g. in this filter not nullable case)?

Reply:
Ok, maybe I have thought too much. It depends on how underlying engines implement but at least such outer join removal and change to inner join has no harm.

Reply:
By changing outer join to inner join, we get the chance to swap the inner and outer relation, then we may be able to build hash table on smaller side, which is a big advantage.

Reply:
That's a good point. Thanks for your insight.